File: 08_pizza_party/ruby/pizza_party.rb
class PizzaParty
  def initialize
    get_stats
    print_stats
  end

  def get_stats
    puts "How many people?"
    @people = gets.chomp.to_i

    puts "How many pizzas?"
    @pizzas = gets.chomp.to_i

    puts "How many slices do the pizzas have?"
    @slices_per_pizza = gets.chomp.to_i
  end

  def total_slices
    @pizzas * @slices_per_pizza
  end

  def slices_per_person
    total_slices / @people
  end

  def remaining_slices
    total_slices % @people
  end

  def print_stats
    puts "#{@people} people with #{@pizzas} pizzas."
    puts "Each person gets #{slices_per_person} pieces of pizza."
    puts "There are #{remaining_slices} leftover."
  end
end

PizzaParty.new

File: python/08_pizza_party/pizza_party.py
class PizzaParty(object):
    def __init__(self):
        self.people = int(input("How many people? "))
        self.pizzas = int(input("How many pizzas do you have? "))
        self.slices_per_pizza = int(input("How many slices do the pizzas have? "))
        self.print_stats()

    def total_slices(self, pizzas, slices_per_pizza):
        return pizzas * slices_per_pizza

    def slices_per_person(self, total_slices_output, people):
        return total_slices_output / people

    def remaining_slices(self, total_slices_output, people):
        return total_slices_output % people

    def print_stats(self):
        print("{} people with {} pizzas.".format(self.people, self.pizzas))

        per_person_slices = self.total_slices(self.pizzas, self.slices_per_pizza)

        print("Each person gets {} pieces of pizza.".format(
            self.slices_per_person(per_person_slices, self.people)))

        print("There are {} slices leftover.".format(
            self.remaining_slices(per_person_slices, self.people)))

party = PizzaParty()
